tests/test_modeling_bert.py

```diff
@@ -260,6 +260,65 @@ def create_and_check_model_for_causal_lm_as_decoder(
         )
         self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
 
+    def create_and_check_decoder_model_past_large_inputs(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+        encoder_hidden_states,
+        encoder_attention_mask,
+    ):
+        config.is_decoder = True
+        config.add_cross_attention = True
+        model = BertLMHeadModel(config=config).to(torch_device).eval()
+
+        # first forward pass
+        outputs = model(
+            input_ids,
+            encoder_hidden_states=encoder_hidden_states,
```

Comment:
think `input_mask` should be passed here as well no? Kinda weird that this test passes actually

Reply:
the  `prepare_inputs_for_generation` method also created mask if it's not passed, so the model always receives a mask, when `use_cache` is `True`

Reply:
Actually I forgot it in Bart as well that's why :D -> I added it in T5 and GPT2 though. Could you fix it in `tests/test_modeling_bart.py` as well please. 

Maybe the attention mask is all ones which is why it passes...